src/authenticationinapp/authenticationinapp.cpp

```diff
@@ -59,8 +67,8 @@ void AuthenticationInApp::registerSession(AuthenticationInAppSession* session) {
 
   m_session = session;
   connect(session, &QObject::destroyed, this, [this]() {
-    m_session = nullptr;
     setState(StateInitializing);
+    m_session = nullptr;
```

Comment:
Oh... this can generate a bug. SetState emits a signal, and we can have code that creates a new session. If that happens, we end up nullifying the new one instead of the old one.

Reply:
Reverted the change.